hooks/pre/31-mongo_storage_engine.rb

```diff
@@ -0,0 +1,67 @@
+def fail_and_exit(message)
+  Kafo::Helpers.log_and_say :error, message
+  kafo.class.exit 1
+end
+
+def migration
+  export_dir = '/var/tmp/mongodb_engine_upgrade'
+  mongo_dir = '/var/lib/mongodb'
+  hiera_file = '/etc/foreman-installer/custom-hiera.yaml'
+  mongodb_backup = '/var/tmp/mongodb_backup'
+  mongo_conf = '/etc/opt/rh/rh-mongodb34/mongod.conf'
+  pulp_db_param = param('katello', 'pulp_db_name')
+
+  # Create export directory and dump MongoDB
+  logger.info 'Stopping Pulp services except MongoDB'
+  Kafo::Helpers.execute('foreman-maintain service stop --exclude "rh-mongodb34-mongod","postgresql","tomcat","dynflowd","foreman-proxy","puppetserver"')
+  FileUtils.mkdir(export_dir)
+  File.chmod(0700, export_dir)
+  logger.info "Starting mongodump to #{export_dir}"
+  Kafo::Helpers.execute("mongodump --host localhost --out #{export_dir}")
+
+  # Move datafiles out of MongoDB directory so it will start
+  logger.info 'Export done, stopping MongoDB to move old datafiles'
+  Kafo::Helpers.execute('foreman-maintain service stop --only rh-mongodb34-mongod')
+  logger.info "Moving contents from #{mongo_dir} to #{mongodb_backup}"
+  FileUtils.mkdir_p(mongodb_backup)
+  File.chmod(0700, mongodb_backup)
+  Kafo::Helpers.execute("mv #{mongo_dir}/* #{mongodb_backup}")
+
+  # Import the dump, fail and notify user of backup if restore does not work.
+  logger.info 'Changing config to WiredTiger and starting restore.'
+  Kafo::Helpers.execute("sed -i.bak -e 's/mmapv1/wiredTiger/g' #{mongo_conf}")
+  Kafo::Helpers.execute("mv #{mongo_conf}.bak #{mongodb_backup}")
+  Kafo::Helpers.execute('foreman-maintain service start --only rh-mongodb34-mongod')
+  if katello
+    Kafo::Helpers.execute("mongorestore --host localhost --db=#{pulp_db_param.value} --quiet --dir=#{export_dir}")
+  else
+    Kafo::Helpers.execute("mongorestore --host localhost --db=pulp_database --quiet --dir=#{export_dir}")
+  end
```

Comment:
Remind me why we need two different conditions here? Whats the situation where `pulp_db_param` is set?

Reply:
That is the name of the pulp db, since we dont support remote db on proxy yet the db name is pulp_database like you suggested below.

Reply:
Can we make https://github.com/Katello/katello-installer/pull/639/files#diff-a5f32e65ed1bfa7ffadaa017de91d6b0R13 smth like `pulp_db_param = param('katello', 'pulp_db_name') || 'pulp_database'`, so that we don't have this `if/else` here?

Reply:
`&.` is a modern ruby construct that will fail on EL7.

Reply:
Right. So while I still think this could be written "nicer", I don't think we should block on this, as it seems to work just fine.

Reply:
Indeed, it could still be written as `pulp_db = katello ? param('katello', 'pulp_db_name').value || 'pulp_database'`